app/models/publisher.rb

```diff
@@ -79,13 +79,6 @@ def wallet
 
     # if the wallet call fails the wallet will be nil
     if @_wallet
```

Comment:
Would you add early termination like `return nil if @_wallet.nil?` to not have nested if statements?